controller/src/beerocks/master/son_master_thread.cpp

```diff
@@ -544,11 +550,14 @@ bool master_thread::autoconfig_wsc_add_m2(std::shared_ptr<ieee1905_1::tlvWscM1>
         return false;
     if (!tlvWscM2->set_serial_number("prpl12345"))
         return false;
-    std::memset(tlvWscM2->uuid_r_attr().data, 0xee, tlvWscM2->uuid_r_attr().data_length);
+    tlvWscM2->primary_device_type_attr().sub_category_id = WSC::WSC_DEV_NETWORK_INFRA_GATEWAY;
+
+    // TODO Maybe the band should be taken from bss_info_conf.operating_class instead?
```

Comment:
According to the specification - the paragraph above Table 44 in section 12 of the WSC technical specification - the RF bands attribute "can be used in the to indicate a specific (or group) of RF Bands to which a setting applies".

So my understanding is that we should set it to `0x7` when bss_info_conf is not set (after dev_reset_default or first boot), and according to bss_info_conf otherwise.

> RF

Reply:
> According to the specification - the paragraph above Table 44 in section 12 of the WSC technical specification - the RF bands attribute "can be used to indicate a specific (or group) of RF Bands to which a setting applies".

No, that sentence only applies to the optional RF Bands attribute in a Credentials/EncryptedData attribute. This is the global (required) RF Bands attribute in the M1/M2. So the relevant sentence is "This attribute is used to indicate a specific RF band that is utilized during message exchange to permit end points and proxies to communicate over a consistent radio interface."


> So my understanding is that we should set it to `0x7` when bss_info_conf is not set (after dev_reset_default or first boot), and according to bss_info_conf otherwise.

Based on the sentence above, it should be set to 0x0 (because no specific RF band is utilized during message exchange). Or maybe, just set it to the same value as M1. 

The BBF implementation sets it to the radio's band in M1, and according to bss_info_conf in M2, and ignores it when receiving M2.

hostapd/wpa_supplicant sets it according to the channel over which the M1 was received. Not much we can do with that...

I'm going to leave it as a TODO :-)